docs/modules/ROOT/pages/udc.adoc

```diff
@@ -0,0 +1,228 @@
+= Universal Deployer Contract
+
+Account contracts make up very critical components of the StarkNet ecosystem, since a bug in any implementation—let alone a widespread one—could prove disastrous.
+Therefore, maximum caution is in order.
+In this spirit, we should consider trimming account responsibilities to simplify implementations and minimize bug/attack surfaces.
+
+This standard moves contract deployment responsibilities to an external, specialized deployer contract.
+The separation allows accounts to deploy contracts without compromising security.
+To remove the redundancy of deploying a new deployer contract for each account, the deployer contract should be a singleton Universal Deployer Contract (UDC).
+
+TIP: For further information on design decisions, see the https://community.starknet.io/t/universal-deployer-contract-proposal/1864[Universal Deployer Contract Proposal].
+
+== Table of Contents
+* <<quickstart, Quickstart>>
+* <<deploying_the_udc,Deploying the UDC>>
+* <<usage,Usage>>
+ ** <<reserved_address_space_for_deployer,Reserved Address Space for Deployer>>
+ ** <<reproducible_deployments,Reproducible Deployments>>
+* <<calculating_counterfactual_addresses,Calculating Counterfactual Addresses>>
+* <<universal_deployer_contract_api,Universal Deployer Contract API>>
+
+== Quickstart
+
+Account contracts need to first declare the contract and then send a `deployContract` transaction to the Universal Deployer Contract address.
+Here's an example implementation in Python:
+
+[,python]
+----
+signer = MockSigner(PRIVATE_KEY)
+
+async def deploy_contract():
+    # Declare the contract and get the class hash before deployment
+    class_hash, _ = await signer.declare_class(account, "Ownable")
+
+    # Set up deployment parameters
+    salt = 123456  # random number
+    unique = FALSE
+    calldata = [account.contract_address]
+    params = [class_hash, salt, unique, len(calldata), *calldata]
+
+    # Send a "deployContract" transaction to the deployer contract
+    await account.send_transaction(
+        account,
+        deployer_address,
+        "deployContract",
+        params
+    )
+----
+
+TIP: The UDC address is `0x041a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02bf`.
+Note that this address may change in the future.
+
+=== Deploying the UDC
+
+Though, the Universal Deployer Contract has already been deployed, this documentation does not take its deployment for granted.
+The deployment of the UDC requires a contract (most likely an account) to call https://www.cairo-lang.org/docs/hello_starknet/deploying_from_contracts.html#the-deploy-system-call[StarkNet's deploy syscall].
```

Comment:
I would add at least a warning, this feature is deprecated and will be removed soon.